### PR TITLE
Support Mageplaza gift card

### PR DESCRIPTION
**Pull request by johnmccombs1**
Asana task: https://app.asana.com/0/564264490825835/1121586614366893

### CODE REVIEW

File: Model/Api/DiscountCodeValidation.php [hunk 1]
```diff
@@ -282,6 +282,12 @@ public function validate()
                 $giftCard = $this->discountHelper->loadAmastyGiftCard($couponCode);
             }
 
+            // Apply Mageplaza_GiftCard
+            if (empty($giftCard)) {
+                // Load the gift card by code
+                $giftCard = $this->discountHelper->loadMageplazaGiftCard($couponCode);
+            }
+
```
Comment (danac-gs) [inline Model/Api/DiscountCodeValidation.php]: You should add a section below in applyingGiftCardCode method where this $giftCard is applied to quote / immutable quote. Every gift cart module has it's own logic.

Comment (johnmccombs1) [reply Model/Api/DiscountCodeValidation.php]: I've updated it.

File: Helper/Cart.php [hunk 2]
```diff
@@ -166,7 +166,9 @@ class Cart extends AbstractHelper
         Discount::GIFT_CARD_ACCOUNT => '',
         Discount::UNIRGY_GIFT_CERT => '',
         Discount::AMASTY_GIFTCARD => 'Gift Card ',
-        Discount::GIFT_VOUCHER => ''
+        Discount::GIFT_VOUCHER => '',
```
Comment (ethanwayda22) [inline Helper/Cart.php]: The code seems to be duplicated here

**Comment by johnmccombs1** [comment]
> @johnmccombs1
> The code looks OK. However, the part regarding applying GC via Bolt checkout still does not work. At least in the latest M2 version, 2.3.1.

I've set up a new Magento 2.3.1 instance, tested, and confirmed it works fine, see the video [here](https://drive.google.com/file/d/1IjTOGOvVb1H7TF8t3Jjl1n6VY3woS6Ac/view). You can test it to confirm on this site https://m231.bolt.devzn.com/with the admin credentials below:
> https://m231.bolt.devzn.com/admin
> coredev1/Coredev12

**Pull request merged**